lib/Runtime/Library/JavascriptMap.cpp

```diff
@@ -64,6 +64,15 @@ namespace Js
 
         Var iterable = (args.Info.Count > 1) ? args[1] : library->GetUndefined();
 
+        if (mapObject->map != nullptr)
+        {
+            JavascriptError::ThrowTypeErrorVar(scriptContext, JSERR_ObjectIsAlreadyInitialized, _u("Map"), _u("Map"));
+        }
+
+        // ensure mapObject->map is created before trying to fetch the adder function
```

Comment:
I would be more explicit in the comment here- specifically, point out that subsequent operations (like fetching the adder) can access the map so all it's fields need to be initialized before these operations can be invoked.